### DISCUSSION TITLE
Next.js Serverless function static directory read

### DISCUSSION THREAD

**Discussion by slavagoreev** [vercel/vercel#8495] [Help] [answered]
Currently, I'm facing an issue `ERROR	Error: ENOENT: no such file or directory, scandir`
I've tried multiple approaches but none of them made a trick for me. I need to serve mdx files as an API. To do this, I've made a simple API that transforms [...slug] into a path inside my `apps/marketing/assets/posts/...` directory. For instance, when I go to `/api/posts/internal/en/`, I should have a list of files in a directory `apps/marketing/assets/posts/internal/en`.

The problem is that those files are not accessible inside a serverless function environment (only `/var/task/build/apps/marketing/serverless/` is there but nothing else). They are only accessible during a build time. *I guess the problem in my monorepo setup*. I've tried to trick @vercel/nft in a various ways using `require`/`readDirSync`

#### I've tried `includeFiles` approach
```json
"functions": {
  "apps/marketing/pages/**": {
    "includeFiles": "apps/marketing/assets/posts/**"
  }
}
  ```
  
#### I've tried to use `fs.readFileSync/fs.readdirSync` approach
```tsx
try {
  readdirSync(resolve(process.cwd(), './assets/posts'));
} catch (e) {
  console.error(e);
}
```

#### I have an extension for `next.config.js`
```js
{
  pageExtensions: ['mdx', 'md', 'jsx', 'js', 'tsx', 'ts'],
  experimental: { nftTracing: true },
}
```

#### Still no luck solving this issue. 
Could you please recommend me something to continue.

<img width="1138" alt="image" src="https://user-images.githubusercontent.com/4031952/187764230-1a77ea58-48b1-434e-b5c6-623a1d6c6ffa.png">

**Comment by amyegan** [comment]
Hi @slavagoreev! You should be able to read the files using `fs.readdirSync` and `fs.readFileSync`. Does your app work locally if you run it with [`vercel dev`](https://vercel.com/docs/cli#commands/dev)? It seems like there might be a problem with the path you're providing to `readdirSync`. We have some examples in our "[How can I use files in Serverless Functions?](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions)" guide.

If you can share your repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I can take a closer look into what might be causing the problem for you.

**Reply by slavagoreev** [reply]
The same functionality worked just fine for a newly created project. Thats weird because the code is exactly the same...
Maybe I should update my project to the newest version of Next.js or reset all the overrides in a project and use a default setting for output directory

**Comment by slavagoreev** [comment, marked as answer]
**I've updated my next.js project to v12 and it made a trick**

**Reply by slavagoreev** [reply]
There is a viable example app of using Serverless function in a monorepo context with next.js
https://github.com/slavagoreev/monorepo-vercel-demo

**Reply by amyegan** [reply]
Thank you for coming back to share this. I'm glad you got it working!
